### ISSUE TITLE
usps "software version is no longer supported"

### ISSUE THREAD

**Issue by bsdlp** [jat255/PackageMate#19]
adding a usps package gives me this error in the status column:

```
This Software version is no longer supported.  You must go to the SA web site (www.uspswebtools.com/Shippingassistant) and download the latest version of the Shipping Assistant.
```

**Comment by jat255** [comment]
Hmm, I can't reproduce that one. I just tried adding a USPS package on my install (using my existing API key) and it worked as expected. 

If you can reproduce the API call that's made here: https://github.com/jat255/PackageMate/blob/master/api_server/3rd-party/trackers.js#L5-L17 with cURL or Postman or something, that might provide some more insight into why it's not working. I'm assuming you set the `USPS_USERNAME` environment variable appropriately?

**Comment by bsdlp** [comment]
yep i replicated the results by manually calling the api with my username

as for username - this is the username for my account on usps.com/developer.usps.com right?

**Comment by jat255** [comment]
> as for username - this is the username for my account on usps.com/developer.usps.com right?

Now that I'm looking at it, I think it's a different thing. The developer portal didn't exist (I don't think) when I first wrote this, so I followed the instructions here:

https://www.usps.com/business/web-tools-apis/track-and-confirm-api.htm#_Toc120028072

That includes a link to register for the web tools access at this address: https://registration.shippingapis.com/

I registered in Sept. 2020, and they sent me an email with a username and password that they create. I haven't had to use the password for anything, just the username.

**Comment by bsdlp** [comment]
great - i think that's the right track. i registered for an account there. without access to the track api, i now get `API Authorization failure. User <userid> is not authorized to use API TrackV2`. I've filed a support ticket as per their email to get access so i presume i'll have things working whenever they get back to me
 
thanks for your help!

**Comment by mdarl24** [comment]
> great - i think that's the right track. i registered for an account there. without access to the track api, i now get `API Authorization failure. User <userid> is not authorized to use API TrackV2`. I've filed a support ticket as per their email to get access so i presume i'll have things working whenever they get back to me
> 
> thanks for your help!

Have they given you access?  I sent in a ticket 2 months ago and have followed up, and I haven't gotten any response beyond the first email saying they don't automatically give that access anymore and give us these 10-15 or so pieces of information to ask for it.

**Comment by jat255** [comment]
@bsdlp and @mdarl24 I've been trying to migrate the USPS scraper to use their newer API, but have not been successful getting any sort of support, either at their stated email support address, or via Github issue (https://github.com/USPS/api-examples/issues/28), so unfortunately I think the old way will remain the only possible option for the foreseeable future. 🫤

**Comment by sochartgit** [comment]
so this project is still alive or ?

**Comment by jat255** [comment]
@sochartgit It's as alive as you want it to be? The original code works and I use it whenever I'm receiving a package. My project isn't exactly responsible for USPS not giving access to/supporting their API...

**Comment by jayem5** [comment]
@jat255 thanks for creating this! Do you have any suggestions as to how to get USPS to function? They're now providing a 'consumer key' and 'consumer secret' however that's resulting in an authorization failure in PackageMate.

**Comment by jat255** [comment]
@jat255 If they've updated their API access procedures, I may take another look at this. When I first tried a while back, I could not even get access to the API and their email and github accounts were totally unresponsive, so I'll try to take another swing at it in the near future. The authorization methods have probably changed, so the existing API requests in packagemate probably won't work, I'd guess
